### DISCUSSION TITLE
How to modify a property of child items in ItemsControl

### DISCUSSION THREAD

**Discussion by dbriard** [AvaloniaUI/Avalonia#11310] [Q&A] [answered]
Hi, 
That's probably not complicated but I can't figure how to make it work:

I have a separator control named FormSeparator which is generally horizontal.
But when this FormSeparator is inside another control named FormGroup (a ItemsControl), I want to change its Orientation property to Vertical... 

I try to use highlighted style in Grid, in ItemsPresenter and in the DockPanel with no success...

![image](https://github.com/AvaloniaUI/Avalonia/assets/3595443/4b265e61-2c7c-4525-ba75-c010e482edcc)

Bonus question: 
I have a hardcoded Spacing=8 on the DockPanel which is used as ItemsPanel.

On WPF I would replace the ItemsPanel by DockPanel with IsItemHost="True" so that I can bind the Spacing property with TemplateBinding, is there a way to do that with Avalonia? 

Thank you.

**Comment by timunie** [comment, marked as answer]
Most likely you set the Orientation locally anywhere else. In that case, a Style cannot override it. https://docs.avaloniaui.net/docs/styling/troubleshooting#locally-set-properties-override-styles

------------

> Bonus question:
> I have a hardcoded Spacing=8 on the DockPanel which is used as ItemsPanel.

> On WPF I would replace the ItemsPanel by DockPanel with IsItemHost="True" so that I can bind the Spacing property with TemplateBinding, is there a way to do that with Avalonia?

Why not using a DynamicResource? this can be adjusted easily. Otherwise you need a "normal" Binding with the `$parent[]` snytax. Should also work I guess.

**Reply by dbriard** [reply]
@timunie 

> Why not using a DynamicResource? this can be adjusted easily. Otherwise you need a "normal" Binding with the $parent[] snytax. Should also work I guess.

That works, I forgot about that, very convinient, thanks!
`ColumnSpacing="{Binding $parent[local:FormGroup].Spacing}"`


> Most likely you set the Orientation locally anywhere else. In that case, a Style cannot override it.
I don't think so.

Orientation is not set in control code, nor in control theme.
```
/// <summary>
    /// Defines the <see cref="Orientation"/> property
    /// </summary>
    public static readonly StyledProperty<Orientation> OrientationProperty =
        AvaloniaProperty.Register<FormSeparator, Orientation>(nameof(Orientation));
```

I would like set automatically switch separator's orientation to Vertical when used like this:
```
<local:FormGroup Header="FormGroup">
                            <local:Button Content="Button" />
                            <local:FormSeparator />
                            <local:Button Content="Button" />
                        </local:FormGroup>
```

**Reply by timunie** [reply]
still think you need to set the reveresed / "normal" style with a different orientation. But not 100 % sure.
